### PR TITLE
maintenance window fix

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
Mobile maintenance windows used to take the minimum start time and max end time of all relevant windows for a service. This caused windows to show as much longer then they actually were. This PR takes the earliest window that's still in the future. 
## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6925

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *Describe the tests completed and the results*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/app/models/mobile/v0/service_graph.rb
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/service_graph.rb:35]: don't love that all these conditions are all together in one if statement but rubocop didn't like it any other way.

File: modules/mobile/app/models/mobile/v0/service_graph.rb
Comment (kpethtel) [inline modules/mobile/app/models/mobile/v0/service_graph.rb:32]: Am I crazy, or is this while loop just an each loop in disguise?

File: modules/mobile/app/models/mobile/v0/service_graph.rb
Comment (kpethtel) [reply modules/mobile/app/models/mobile/v0/service_graph.rb:35]: I think this might be slightly clearer with `dss.present?` instead of `!dss`. Dealing with the inversion within a long series of conditions is harder for the brain to wrap around.

File: modules/mobile/spec/request/maintenance_windows_request_spec.rb
Comment (kpethtel) [inline modules/mobile/spec/request/maintenance_windows_request_spec.rb:131]: I think these tests would be slightly clearer for future readers if we instead created these windows to variables (idk if these variable names are accurate):
```
        let!(:earliest_starting) { FactoryBot.create(:mobile_maintenance_evss_first) }
        let!(:middle_starting) { FactoryBot.create(:mobile_maintenance_evss_second) }
        let!(:latest_starting) { FactoryBot.create(:mobile_maintenance_evss_third) }
```

Then your tests can test again those values like:
`expect(attributes.pluck('startTime').uniq).to eq([earliest_starting.start_time])`

It makes the tests more self-documenting.

**Review by kpethtel — DISMISSED** [review]
Couple suggestions, but nothing blocking. 👍

File: modules/mobile/spec/request/maintenance_windows_request_spec.rb
Comment (aherzberg) [reply modules/mobile/spec/request/maintenance_windows_request_spec.rb]: sounds good. updated

File: modules/mobile/app/models/mobile/v0/service_graph.rb
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/service_graph.rb:32]: haha yeah, even tho it's ugly I didn't wanna touch it in fear of introducing some differing logic since that's not technically what's wrong with the code.

File: modules/mobile/app/models/mobile/v0/service_graph.rb
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/service_graph.rb]: sounds good. updated to use `nil?`

**Review by kpethtel — APPROVED** [review]
👍

**Review by jperk51 — APPROVED** [review]

**Comment by aherzberg** [comment]
@jperk51 @kpethtel forgot about some other tests that got broken when changing up test data.

**Review by kpethtel — APPROVED** [review]

**Comment by jperk51** [comment]
Pls don't merge this before the daily prod deploy. @aherzberg reach out to @rmtolmach to set up a staging test like we did previously so we can test quickly after this is merged with plenty of time to rollback before the next prod deploy

**Pull request merged**
